src/content/drafts/sourcecode-to-tokens-AST.md

```diff
@@ -0,0 +1,103 @@
+
+:information_source:**Tip**: If you ever get lost in the AST, you can hover over different sections of your source code on the right and the tool then highlights the relevant node in the AST on the left.
+
+The AST represents the relationship between the tokens generated from the source code as a tree comprising different nodes. Every node has at least a type specifying what it is representing. For example, a node type could be a `Literal` representing an actual value or a `CallExpression` that represents a function call. The `Literal` node might only contain a value whereas the `CallExpression` node can contain additional information such as what is being called (`callee`), or what `arguments` are being passed in.
+
+:memo: It is important to remember that the format of AST changes depending on the type of programming language and the type of tool used for parsing. 
+
+###  Linting using AST
+You can traverse the different nodes of the AST to gain insights or perform actions. One of the most common use cases that extensively use AST traversal is
+[ linting](https://en.wikipedia.org/wiki/Lint_(software)). ESLint uses [Espree](https://github.com/eslint/espree)  parser to create an AST from JavaScript code and the AST is defined by the  common standard [ESTree](https://github.com/estree/estree) specification. To understand the different types of nodes represented in the AST refer to [ES5 AST grammar documentation](https://github.com/estree/estree/blob/master/es5.md). You can refer to ESLint[ documentation](https://eslint.org/docs/latest/user-guide/configuring/rules#disabling-rules) to learn more about their built-in rules and how you can build custom rules, plugins, and formatters.
+
+Here's an[ example in the AST explorer](https://astexplorer.net/#/gist/052d25ec2db5e45442e474ffc8dd0f43/f2146de2d27649296a810fc4e88293b649782649) that establishes a rule that you cannot have an empty catch block. We then traverse the AST and write a rule in ESLint to check if there are empty catch blocks in the source code.
+```
+"use strict";
+module.exports = {
+  meta: {
+    type: 'problem', // `problem`, `suggestion`, or `layout`
+    docs: {
+      description: "This rule catches an empty catch block",
+       },
+    fixable: null, // Or `code` or `whitespace`
+    schema: [], // Add a schema if the rule has options
+    messages: {
+      emptyCatch: 'Empty catch block is not allowed.',
+  },
+  },
+  create(context) {
+    return {
+      CatchClause(node)
+        {
+          if (node.body.body.length === 0) {
+            context.report({node: node.body, messageId: 'emptyCatch'});
+          }
+        }
+    };
+  },
+};
+
+```
+
+In the above code snippet, we check if the length of the body of the `CatchClause` node is 0. If yes,  ESLint’s [report](https://eslint.org/docs/latest/developer-guide/working-with-rules#contextreport) method is used to raise a complaint.
+
+## Step 3: Code generation
+Need inputs from engineers.
```

Comment:
I think we can skip code generation. It's not important for ESLint.

Reply:
Fixed.